### DISCUSSION TITLE
Entities and Database Schema

### DISCUSSION THREAD

**Discussion by LeoBorai** [whizzes/linx#63] [General]
In order to introduce Gardens on a easy to follow structure I have been thinking about the following DB Structure.
A User have the capability to create `Post`s which other users can comment on.
A Comment can be turned into `Post`s instances which at the same time have the same capabilities.

Every `Post` has:

- Rich Text using EditorJS
- Comments
- Can be found via exploring

Comments allow for following up in the same topic, and as these grow in number they are likely to end
up as a new `Post`, a user could read through comments and determine the origin of a topic in a comment
which turned into a `Post`.

We need a way to keep all media (images, audios and videos), to achieve this we can use MinIO and introduce
files (objects) to the ecosystem via `Media` table.

> [`Pxid`s are Prefixed Unique Identifiers](https://github.com/whizzes/pxid) we can use to easily understand relations when working with the API.

<img width="863" alt="Screenshot 2023-07-01 at 3 34 27 PM" src="https://github.com/whizzes/gabble/assets/34756077/caad6eee-96b1-45c9-ab89-c9619e5d7fed">

<details>
    <summary>QDBD File</summary>

Visit: https://app.quickdatabasediagrams.com/#/

```qdbd
# Modify this code to update the DB schema diagram.
# To reset the sample schema, replace everything with
# two dots ('..' - without quotes).

User
-
Id PK Pxid
Name Varchar(64)
Surname NULLABLE Varchar(64)
Username Varchar(64)
Email Varchar(64)
CreatedAt Timestamp
UpdatedAt Timestamp

Media
-
Id PK Pxid
AuthorId FK - User.Id Pxid
StorageId Varchar(?)
MimeType Varchar(32)
CreatedAt Timestamp
UpdatedAt Timestamp

Post
-
Id PK Pxid
AuthorId FK - User.Id Pxid
Title Varchar(256)
Content JSON
CreatedAt Timestamp
UpdatedAt Timestamp

Comment
-
Id PK Pxid
AuthorId FK - User.Id Pxid
PostId FK - Post.Id Pxid
Content Varchar(256)
CreatedAt Timestamp
UpdatedAt Timestamp
```
</details>

**Comment by Michael-Liendo** [comment]
Hello everyone

I have seen the suggestions in Discord server and I would like to propose the following changes in the database schema:

- Remove the `comment` entity and its FKs/Relation fields in other entities
- Provide a `Head` field for Post with type BOOLEAN.

This is what the database would look like:
![image](https://github.com/whizzes/gabble/assets/70660410/4d3b6da6-beaa-4c22-809a-e258d034580c)


These changes were suggested by Ahq, and @CudiLala. 
I believe this proposal will help simplify the schema and treat both posts and comments similarly, improving the structure and making queries easier in the future.

**Reply by LeoBorai** [reply]
Great @Michael-Liend! I think we can go with this on https://github.com/whizzes/gabble/pull/62! Do you want to update it with these changes?

**Reply by Michael-Liendo** [reply]
Yeah! Sure 🚀
